frontend/src/metabase/home/components/HomePage/HomePage.tsx

```diff
@@ -1,31 +1,51 @@
 import { useEffect } from "react";
 import { replace } from "react-router-redux";
 import { isSmallScreen } from "metabase/lib/dom";
-import { useDispatch } from "metabase/lib/redux";
+import { openNavbar } from "metabase/redux/app";
+import { useDispatch, useSelector } from "metabase/lib/redux";
+import {
+  useDatabaseListQuery,
+  useSearchListQuery,
+} from "metabase/common/hooks";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+import { canUseMetabotOnDatabase } from "metabase/metabot/utils";
+import { CollectionItem } from "metabase-types/api";
+import Database from "metabase-lib/metadata/Database";
+import {
+  getCustomHomePageDashboardId,
+  getIsMetabotEnabled,
+} from "../../selectors";
+import { HomeLayout } from "../HomeLayout";
+import { HomeContent } from "../HomeContent";
 
-import HomeLayout from "../HomeLayout";
-import HomeContent from "../../containers/HomeContent";
+const SEARCH_QUERY = { models: "dataset", limit: 1 } as const;
 
-export interface HomePageProps {
-  hasMetabot: boolean;
-  homepageDashboard?: number;
-  onOpenNavbar: () => void;
-}
-
-const HomePage = ({
-  hasMetabot,
-  onOpenNavbar,
-  homepageDashboard,
-}: HomePageProps): JSX.Element => {
+export const HomePage = (): JSX.Element => {
+  const { data: databases, ...databaseState } = useDatabaseListQuery();
+  const { data: models, ...modelState } = useSearchListQuery({
+    query: SEARCH_QUERY,
+  });
+  const isLoading = databaseState.isLoading || modelState.isLoading;
+  const error = databaseState.error ?? modelState.error;
+  const dashboardId = useSelector(getCustomHomePageDashboardId);
+  const isMetabotEnabled = useSelector(getIsMetabotEnabled);
+  const hasMetabot = getHasMetabot(databases, models, isMetabotEnabled);
   const dispatch = useDispatch();
+
   useEffect(() => {
     if (!isSmallScreen()) {
-      onOpenNavbar();
+      dispatch(openNavbar());
+    }
+  }, [dispatch]);
+
+  useEffect(() => {
+    if (dashboardId) {
+      dispatch(replace(`/dashboard/${dashboardId}`));
     }
-  }, [onOpenNavbar]);
+  }, [dashboardId, dispatch]);
```

Comment:
If we "lift" this code up we can avoid rendering the `HomePage` component, which will avoid triggering `useDatabaseListQuery` & `useSearchListQuery`, which will save 2 network requests + time spent on rendering. 

I believe the best way to do it would be to move this redirect to [`onEnter` of `/` Route in `routes.jsx`](https://github.com/metabase/metabase/blob/831b532/frontend/src/metabase/routes.jsx#L124-L129).

Seems worth it given that it's the entry page of the app.

Reply:
These queries are used across the app, they is no harm is using them here. Furthermore, it somewhat improves the performance of other pages because they get preloaded (not intentionally, obviously).

Reply:
I might have done it differently if I were implementing the feature, but here I simplify update the existing code without changing anything.